### ISSUE TITLE
when color.alternate is set to change the background, bold is always removed from text on alternate rows

### ISSUE THREAD

**Issue by bgravato** [GothenburgBitFactory/taskwarrior#2666]
Running taskwarrior 2.6.1.

When color.alternate is set to change the background with "on _something_", bold is removed from the rows that have the alternate background.

For example: `color.alternate=on gray5`

Even if I set bold on color.alternate (example: `color.alternate=bold magenta on gray5`) it will not be bold on the row with the gray5 background. In the give example text color will change to magenta (when no other higher priority color applies), but it won't turn bold. Lines that would  normally be bold, will have bold removed on the lines with gray5 background.

If I don't set "on _something_" in color.alternate (for example: `color.alternate=magenta`), then bold will be preserved.

If I set bold for alternate (eg. `color.alternate=bold magenta`), alternate rows will be bold (and magenta).

To sum up: if background is not set in color.alternate it will work as expected, but if "on _somecolor_" is set, bold is always removed from alternate rows.

**Comment by xjjak** [comment]
The problem are the 256- and 16-color spaces, see issue [[TW-585] bold only respects "named" colors #620 ](https://github.com/GothenburgBitFactory/taskwarrior/issues/620)
As stated in the man task-color when mixing 16- and 256-colors:

>   Note also that there is no bold or bright attributes when dealing with 256 colors, but there is still underline available.

The "named" colors are part of the 16-colors so they will work, but not for `gray`, `rgb` or `color`. Try `color.alternate=bold on magenta`, that test should work.
If you use a dark theme, I use the solarized-dark-256 theme, try  black as the background, it worked for me, as it appears as gray.
